arbos/tx_processor.go

```diff
@@ -453,37 +472,53 @@ func (p *TxProcessor) EndTxHook(gasLeft uint64, success bool) {
 		}
 
 		maxRefund := new(big.Int).Set(inner.MaxRefund)
-		refundNetworkFee := func(amount *big.Int) {
-			const errLog = "network fee address doesn't have enough funds to give user refund"
+		refund := func(refundFrom *common.Address, amount *big.Int) {
```

Comment:
nit: This function doesn't really need a pointer, and it derefs it (and thus assumes it isn't nil), so let's just pass in a regular `common.Address` instead and reference it as needed in the function. The reason TransferBalance takes in a pointer is because it might be nil (meaning it would mint funds)